server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -76,43 +79,51 @@ export default class ServiceProviderReferralsController {
-      res.locals.user.token.accessToken,
-      SPDashboardType.MyCases
-    )
-    this.renderDashboard(res, referralsSummary, 'My cases')
+    const pageSize = config.apis.interventionsService.dashboardPageSize.sp.myCases
+    await this.renderDashboard(req, res, { concluded: false, assignedTo: res.locals.user.userID }, 'My cases', pageSize)
   }
 
   async showAllOpenCasesDashboard(req: Request, res: Response): Promise<void> {
-    const referralsSummary = await this.interventionsService.getServiceProviderSentReferralsSummaryForUserToken(
-      res.locals.user.token.accessToken,
-      SPDashboardType.OpenCases
-    )
-    this.renderDashboard(res, referralsSummary, 'All open cases')
+    const pageSize = config.apis.interventionsService.dashboardPageSize.sp.openCases
+    await this.renderDashboard(req, res, { concluded: false }, 'All open cases', pageSize)
   }
 
   async showUnassignedCasesDashboard(req: Request, res: Response): Promise<void> {
-    const referralsSummary = await this.interventionsService.getServiceProviderSentReferralsSummaryForUserToken(
-      res.locals.user.token.accessToken,
-      SPDashboardType.UnassignedCases
-    )
-    this.renderDashboard(res, referralsSummary, 'Unassigned cases')
+    const pageSize = config.apis.interventionsService.dashboardPageSize.sp.unassignedCases
+    await this.renderDashboard(req, res, { concluded: false, unassigned: true }, 'Unassigned cases', pageSize)
   }
 
   async showCompletedCasesDashboard(req: Request, res: Response): Promise<void> {
-    const referralsSummary = await this.interventionsService.getServiceProviderSentReferralsSummaryForUserToken(
-      res.locals.user.token.accessToken,
-      SPDashboardType.CompletedCases
-    )
-    this.renderDashboard(res, referralsSummary, 'Completed cases')
+    const pageSize = config.apis.interventionsService.dashboardPageSize.sp.completedCases
+    await this.renderDashboard(req, res, { concluded: true }, 'Completed cases', pageSize)
   }
 
-  private renderDashboard(
+  private async renderDashboard(
+    req: Request,
     res: Response,
-    referralsSummary: ServiceProviderSentReferralSummary[],
-    dashboardType: DashboardType
-  ): void {
-    const presenter = new DashboardPresenter(referralsSummary, dashboardType, res.locals.user)
+    getSentReferralsFilterParams: GetSentReferralsFilterParams,
+    dashboardType: DashboardType,
+    pageSize: string
+  ) {
+    const pageNumber = req.query.page
+    const paginationQuery = {
+      page: pageNumber ? Number(pageNumber) : undefined,
+      size: Number(pageSize),
+      sort: ['sentAt,DESC'],
+    }
+
+    const cases = await this.interventionsService.getSentReferralsForUserTokenPaged(
+      res.locals.user.token.accessToken,
+      getSentReferralsFilterParams,
+      paginationQuery
+    )
+
+    const dedupedInterventionIds = Array.from(new Set(cases.content.map(referral => referral.referral.interventionId)))
```

Comment:
not a blocker: i could be good to abstract this logic from here and the pp dashboard and put a unit test around it